mubench.reviewsite/src/Models/Misuse.php

```diff
@@ -45,6 +45,9 @@ public function findings()
     public function snippets()
     {
         $snippets = Snippet::of($this->getProject(), $this->getVersion(), $this->misuse_muid, $this->getFile())->get();
+        if($this->run->experiment_id == 2){
+            $snippets = $snippets->where('detector_muid', $this->detector->muid);
```

Comment:
Instead of checking for Ex2, can we query by `(detector_id = $this->detector->muid OR detector_id IS NULL)`?